internal/command/e2etest/provider_dev_test.go

```diff
@@ -18,6 +18,16 @@ import (
 // we normally do, so they can just overwrite the same local executable
 // in-place to iterate faster.
 func TestProviderDevOverrides(t *testing.T) {
+	if terraformBin != "" {
+		// We're running in a separate-build-then-run context, so we can't
+		// current include this test which depends on being able to build
+		// new executable at runtime.
+		// FIXME: We could potentially add a compromise mode where the test
+		// is allowed to assume that a Go toolchain is available, as long
+		// as it only tries to compile files that were included in the testdata
+		// (which is true for this test).
+		t.Skip("can't run without building a new provider executable")
+	}
```

Comment:
Unless I'm misunderstanding something, these skipped e2e tests don't seem to run in CI at all any more. They seem to be skipped both in the new build-then-run actions and in the per-commit CircleCI runs.

If that's the case, is it feasible to fix that before merge?

Reply:
Ahh yes, you are right that the "normal" path which builds a custom binary sneakily updates this variable before it returns:

https://github.com/hashicorp/terraform/blob/535da4ebc754a7b246c3314aeeeceab43b0ac854/internal/command/e2etest/main_test.go#L21-L39

I'll fix this up by adding a separate variable to track whether we're using an externally-provided executable or a locally-built one, and then change these conditions to all depend on that instead, as a more robust temporary solution to this quirk. Thanks!